crates/turborepo-lib/src/shim.rs

```diff
@@ -792,306 +724,149 @@ mod test {
     }
 
     #[test]
-    fn test_process_potential_turbo_roots() {
-        struct TestCase {
-            description: &'static str,
-            infer_infos: Vec<InferInfo>,
-            output: Result<AbsoluteSystemPathBuf>,
-        }
+    fn test_repo_state_infer() {
+        // Directory layout:
+        // <tmp_dir>
+        //   irrelevant/
+        //   monorepo_root/
+        //     package.json
+        //     standalone/
+        //       package.json
+        //     standalone_monorepo/
+        //       package.json
+        //       packages/
+        //         app-2/
+        //     packages/
+        //       app-1/
+        //         package.json
+        //         src/
+        //   single_root/
+        //     package.json
+        //     src/
```

Comment:
Does this setup really reflect any kind of real world scenario or is it trying to fit a bunch of use cases into the same test? I feel like it makes for an overly complicated test 🤔  (especially `standalone_monorepo` inside `monorepo_root`)

Reply:
Definitely a real-world scenario. I have parts of this setup on my machine (a mix of monorepos and single-package repos as siblings in a directory). But yes, this is also somewhat the convenience of reusing the fixture. 

The nested standalone inside a monorepo piece specifically is mostly not really reflective of a real world scenario, although it does occur. It is something we have code to explicitly handle as well as being covered by the test that this test is replacing